testdata/baselines/reference/submoduleAccepted/conformance/errorOnFunctionReturnType.errors.txt.diff

```diff
@@ -1,50 +1,47 @@
+ foo.js(13,5): error TS2322: Type 'string' is not assignable to type 'number'.
 -foo.js(16,60): error TS2355: A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value.
 -foo.js(21,20): error TS2355: A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value.
--foo.js(31,10): error TS2534: A function returning 'never' cannot have a reachable end point.
++foo.js(16,5): error TS2355: A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value.
++foo.js(21,5): error TS2322: Type '() => void' is not assignable to type 'FunctionReturningPromise'.
++  Type 'void' is not assignable to type 'Promise<number>'.
+ foo.js(31,10): error TS2534: A function returning 'never' cannot have a reachable end point.
 -foo.js(35,12): error TS1065: The return type of an async function or method must be the global Promise<T> type.
--foo.js(37,5): error TS2322: Type '"asd"' is not assignable to type 'never'.
+ foo.js(37,5): error TS2322: Type '"asd"' is not assignable to type 'never'.
 -foo.js(40,56): error TS2534: A function returning 'never' cannot have a reachable end point.
 -foo.js(45,18): error TS2534: A function returning 'never' cannot have a reachable end point.
 -
 -
 -==== foo.js (10 errors) ====
-+foo.js(21,5): error TS2322: Type '() => void' is not assignable to type 'FunctionReturningPromise'.
-+  Type 'void' is not assignable to type 'Promise<number>'.
++foo.js(40,5): error TS2534: A function returning 'never' cannot have a reachable end point.
 +foo.js(45,5): error TS2322: Type '() => void' is not assignable to type 'FunctionReturningNever'.
 +  Type 'void' is not assignable to type 'never'.
 +
 +
-+==== foo.js (2 errors) ====
++==== foo.js (8 errors) ====
      /**
       * @callback FunctionReturningPromise
       * @returns {Promise<number>}
-@@= skipped -17, +11 lines =@@
-     
-     /** @type {FunctionReturningPromise} */
-     function testPromise1() {
--             ~~~~~~~~~~~~
--!!! error TS2355: A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value.
-         console.log("Nope");
+@@= skipped -23, +23 lines =@@
      }
      
      /** @type {FunctionReturningPromise} */
 -               ~~~~~~~~~~~~~~~~~~~~~~~~
 -!!! error TS2705: An async function or method in ES5 requires the 'Promise' constructor.  Make sure you have a declaration for the 'Promise' constructor or include 'ES2015' in your '--lib' option.
      async function testPromise2() {
          return "asd";
--        ~~~~~~
--!!! error TS2322: Type 'string' is not assignable to type 'number'.
+         ~~~~~~
+@@= skipped -9, +7 lines =@@
      }
      
      var testPromise3 = /** @type {FunctionReturningPromise} */ function() {
 -                                                               ~~~~~~~~
--!!! error TS2355: A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value.
++        ~~~~~~~~~~~~
```

Comment:
Hm, curious that this error span is on the var when the JSDoc is on the function?

Reply:
`errorNode := core.OrElse(fn.Type(), fn)`, probably should be `fn.Type() || fn.WholeType || fn`